README.md

```diff
@@ -1,113 +1,158 @@
+### 1. The `layoutsubtree` attribute
+The `layoutsubtree` attribute on a `<canvas>` element opts in canvas descendants to have layout and participate in hit testing. It causes the direct children of the `<canvas>` to have a stacking context, become a containing block for all descendants, and have paint containment.
 
-Descendants of a `<canvas>` with `layoutsubtree` are automatically hit testable, and this approach encourages authors to keep their drawn content in sync with the DOM, ensuring accessibility is up-to-date. Although CSS transform on the drawn element is ignored by `drawElementImage`, it is still honored by hit testing, as well as features like accessibility, anchor positioning, and intersection observer. See [Issue#11](https://github.com/WICG/html-in-canvas/issues/11) for an ongoing discussion of accessibility concerns.
+### 2. `drawElementImage` (and WebGL/WebGPU equivalents)
+The `drawElementImage` method renders a DOM element and its subtree into the canvas.
 
-Offscreen canvas contexts and detached canvases are not supported because drawing DOM content when the canvas is not in the DOM poses technical challenges. See [Issue#2](https://github.com/WICG/html-in-canvas/issues/2) for further discussion.
+**Requirements & Constraints:**
+* `layoutsubtree` must be specified on the `<canvas>`.
+* The `element` must be a direct child of the `<canvas>`.
+* **Transforms:** The canvas's current transformation matrix is applied. CSS transforms on the source `element` are **ignored** for drawing (but still apply to hit testing/accessibility, see below).
+* **Clipping:** Overflowing content (both layout and ink overflow) is clipped to the element's content box.
+* **Sizing:** The optional `width`/`height` arguments specify a destination rect in canvas coordinates. If omitted, the element is drawn at its intrinsic size.
 
-```idl
-interface CanvasRenderingContext2D {
+**WebGL/WebGPU Support:**
+Similar methods are added for 3D contexts: `WebGLRenderingContext.texElementImage2D` and `copyElementImageToTexture`.
 
-  ...
+### 3. `fireOnEveryPaint`
+A `fireOnEveryPaint` option is added to `ResizeObserverOptions`. This allows script to be notified whenever descendants of a `<canvas>` may render differently and may need to be re-drawn. The callback runs at Resize Observer timing (after DOM style/layout, but before paint).
 
-  [RaisesException]
-  void drawElementImage(Element element, unrestricted double x, unrestricted double y);
+### Synchronization
+Once drawn, the canvas image is static. Authors must explicitly redraw the element to reflect subsequent changes. To assist with syncing the DOM position with the drawn image (essential for accessibility and hit testing), `drawElementImage` returns a `DOMMatrix` which can be applied to the DOM element's `transform` style to make the layout position match the drawn position. The `getElementTransform` API is a helper to return the same `transform` for WebGL/WebGPU cases.
```

Comment:
How does getElementTransform work? Is that actually possible?

Reply:
I rewrote this to make it clearer. It now says:
> It is essential for hit testing, intersection observer, and hit testing that the DOM position is kept in sync with the position of the drawn image.
> To assist with this, `drawElementImage` returns a `DOMMatrix` which can be applied to the DOM element's `transform` style to make the DOM position match the drawn position. There is no explicit current transform matrix in WebGL/WebGPU, so the `getElementTransform(element, draw_transform)` helper method is provided to make it easy to adjust a provided `DOMMatrix` transform so that it can be applied to the element's `transform` style to make the DOM position match the drawn position.